--- a/cardano-crypto-class.cabal
+++ b/cardano-crypto-class.cabal
@@ -148,9 +148,7 @@
   if impl(ghc <9.0.0)
     build-depends:
       integer-gmp
-  pkgconfig-depends:
-    libblst >=0.3.14,
-    libsodium,
+  build-depends: blst-clib >=0.3.14, sodium-clib
 
   c-sources: cbits/blst_util.c
   include-dirs: cbits
@@ -162,7 +160,7 @@
       Cardano.Crypto.SECP256K1.C
       Cardano.Crypto.SECP256K1.Constants
 
-    pkgconfig-depends: libsecp256k1
+    build-depends: secp256k1-clib
     cpp-options: -DSECP256K1_ENABLED
 
 library testlib
@@ -197,7 +195,7 @@
     cardano-crypto-class,
     cborg,
     containers,
-    contra-tracer ^>=0.2.1,
+    contra-tracer >=0.1.0.2 && <0.3,
     crypton,
     formatting,
     hspec,
@@ -227,66 +225,3 @@
     bytestring,
     cardano-crypto-class,
 
-test-suite tests
-  import: base, project-config
-  type: exitcode-stdio-1.0
-  hs-source-dirs: test
-  other-modules: Paths_cardano_crypto_class
-  autogen-modules: Paths_cardano_crypto_class
-  main-is: Main.hs
-  build-depends:
-    QuickCheck,
-    base,
-    bytestring,
-    cardano-base:testlib >=0.1.3,
-    cardano-crypto-class:{cardano-crypto-class, testlib},
-    hspec,
-
-  other-modules:
-    Test.Cardano.Crypto.PackedBytes.Spec
-
-  if flag(secp256k1-support)
-    cpp-options: -DSECP256K1_ENABLED
-  ghc-options:
-    -threaded
-    -rtsopts
-    -with-rtsopts=-N
-
-benchmark bench
-  import: base, project-config
-  type: exitcode-stdio-1.0
-  hs-source-dirs: bench
-  main-is: Main.hs
-  other-modules:
-    Bench.Crypto.DSIGN
-    Bench.Crypto.HASH
-    Bench.Crypto.KES
-
-  build-depends:
-    base,
-    bytestring,
-    cardano-binary,
-    cardano-crypto-class:{cardano-crypto-class, benchlib},
-    criterion,
-    deepseq,
-
-  if flag(secp256k1-support)
-    cpp-options: -DSECP256K1_ENABLED
-  ghc-options: -threaded
-
-test-suite test-memory-example
-  import: base, project-config
-  -- Temporarily removing this as it is breaking the CI, and
-  -- we don't see the benefit. Will circle back to this to decide
-  -- whether to modify or completely remove.
-  buildable: False
-  type: exitcode-stdio-1.0
-  hs-source-dirs: memory-example
-  main-is: Main.hs
-  build-depends:
-    base,
-    bytestring,
-    cardano-crypto-class,
-
-  if (os(linux) || os(osx))
-    build-depends: unix
